ddtrace/llmobs/utils.py

```diff
@@ -21,6 +21,18 @@
 Message = TypedDict("Message", {"content": str, "role": str}, total=False)
 
 
+class AnnotationContext:
+    def __init__(self, _tracer, _annotation_callback):
+        self._tracer = _tracer
+        self._annotate_prompt = _annotation_callback
+
+    def __enter__(self):
+        self._tracer.on_start_span(self._annotate_prompt)
```

Comment:
> There's a limitation here - we don't know the span kind right when the span is started since this value is set as a tag after-the-fact. This is problematic for Langchain spans, where the span kind may be conditional based on some post-processing logic

Assuming you're talking about `span.span_type`, why is this problematic for langchain spans? I believe `_on_start_span()` hooks run after `tracer._start_span(span_type=...)` has begun.

Reply:
Sorry i mean the span kind (llm, retrieval, embedding, etc.) which is set as a tag, not span kind

Reply:
I'm still a tad confused, why is not knowing `span_kind` a limitation for settings tags in the annotation context? Or are you referring to future considerations when we will add prompt/I/O options to annotation contexts?

Reply:
Yea, i was referring to future considerations when the span kind is necessary. But for the data was need to annotate right now using context (prompt, name, tags) i don't think it is